personalwebsite/projects/admin.py

```diff
@@ -13,12 +16,67 @@ def clean(self):
         cleaned_data = self.cleaned_data
         tags = cleaned_data['tags']
         if len(tags) > 8:
-            raise ValidationError(
-                "You can't assign more than 8 tags to a project.")
+            raise ValidationError("You can't assign more than 8 tags to a project.")
         return cleaned_data
 
 
+class ImageForm(forms.ModelForm):
+    class Meta:
+        model = Image
+        fields = ['name', 'image', 'default', 'album']
+
+    def clean(self):
+        cleaned_data = self.cleaned_data
+        default = cleaned_data['default']
+        try:
+            image_album = ImageAlbum.objects.get(pk=cleaned_data['album'].pk)
+        except AttributeError:
+            return cleaned_data
+
+        # does this album already have a default image set?
+        if image_album and default:
+            for image in image_album.images.all():
+                if image.default:
+                    raise ValidationError("This image album already has a default set.")
+        
+        return cleaned_data
+
+
+
 @admin.register(Project)
 class ProjectAdmin(admin.ModelAdmin):
     form = ProjectForm
     filter_horizontal = ('tags',)
+
+
+@admin.register(Image)
```

Comment:
Image and ImageAlbum can be condensed inside project since they only belong inside projects. I also found it hard to traverse through this in the admin page and ran into a few problems with unique constraints.